### DISCUSSION TITLE
Add cookie parser for custom endpoints

### DISCUSSION THREAD

**Discussion by Thomasparsley** [directus/directus#5059] [Feature Requests]
In custom API endpoint are no parsed cookies in request. But in `package.json` for API between dependencies is `cookie-parser`.

Custom api endpoint:
```javascript
router.get('/', async (req, res) => {
    console.log({ req })

    res.send({ o: 'k' })
})
```

Request:
```javascript
...
header: {
    cookie: 'uuid=bef29d44-896d-4ee8-8430-bf66d8ba75fd',
    ...
}
...
```

When I searched this repository, I found that for [`GraphQL '/'`, `GraphQL '/system'`](https://github.com/directus/directus/blob/119ad0c1ee820a8d1fe3179bbdf04766600a74fd/api/src/controllers/graphql.ts), a cookies parser is inserted as middleware. And the cookie parser is also used for the [auth controller](https://github.com/directus/directus/blob/1b1e500ee9f06711c2edc0560518714f8e4521f0/api/src/controllers/auth.ts).

I was unable to find another use of the cookie parser in the API :(.

I think a cookie parser should also be used here as middleware:
https://github.com/directus/directus/blob/ecd5526b55aaab3b04e20f9e21285df3600b02ca/api/src/extensions.ts#L128

Or if I want to use a cookie parser, do I have to insert it in the custom endpoint as middleware?

Directus 9.0.0-rc.57
DBMS: MariaDB 10.3.25
Deployment: npx
Browser: Chrome 89 - 64x
Node: 14.16.0
Local VM - Ubuntu 20.10

**Comment by rijkvanzanten** [comment]
Yeah I don't think there's an inherent downside to having the cookie parser on the global level, just like the body parser 👍🏻  Wanna open a PR for that?

**Reply by Thomasparsley** [reply]
Having a global cookie parser would be a minimal reduction in performance. Yeah, why not.

**Comment by jaads** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
